easybuild/easyblocks/f/freefem.py

```diff
@@ -0,0 +1,165 @@
+##
+# Copyright 2009-2020 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for FreeFem++, implemented as an easyblock
+
+@author: Balazs Hajgato (Free University Brussels - VUB)
+@author: Kenneth Hoste (HPC-UGent)
+"""
+import os
+import re
+
+import easybuild.tools.environment as env
+import easybuild.tools.toolchain as toolchain
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.modules import get_software_root
+from easybuild.tools.filetools import change_dir
+from easybuild.tools.run import run_cmd
+
+
+class EB_FreeFEM(ConfigureMake):
+    """Support for building and installing FreeFem++."""
+
+    def configure_step(self):
+        """
+        FreeFem++ configure should run twice.
+        First to configure PETSc (and then build it),
+        then to configure FreeFem++ with the built PETSc.
+        """
+
+        # first Autoreconf has to be run
+        if not get_software_root('Autoconf'):
+            raise EasyBuildError("Autoconf is required to build FreeFem++. Please add it as build dependency")
+
+        run_cmd("autoreconf -i", log_all=True, simple=False)
+
+        configopts = [
+            '--disable-optim',  # disable custom optimizations (not needed, $CFLAGS set by EasyBuild is picked up)
+            '--enable-download',  # enable downloading of dependencies
+            '--disable-openblas',  # do not download OpenBLAS
+        ]
+
+        blas_family = self.toolchain.blas_family()
+        if blas_family == toolchain.OPENBLAS:
+            configopts.append("--with-blas=openblas")
```

Comment:
It seems that `--with-blas` expects a path to the libs
```suggestion
            openblas_root = get_software_root('OpenBLAS')                           
            configopts.append('--with-blas=%s' % os.path.join(openblas_root, 'lib'))
```